src/components/tx/modals/TokenTransferModal/ReviewSpendingLimitTx.tsx

```diff
@@ -0,0 +1,134 @@
+import { ReactElement, SyntheticEvent, useMemo, useState } from 'react'
+import { BigNumberish, BytesLike } from 'ethers'
+import { Box, Button, DialogContent, Typography } from '@mui/material'
+import SendFromBlock from '@/components/tx/SendFromBlock'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import { ReviewTokenTxProps, TokenTransferReview } from '@/components/tx/modals/TokenTransferModal/ReviewTokenTx'
+import useBalances from '@/hooks/useBalances'
+import useSpendingLimit from '@/hooks/useSpendingLimit'
+import useSpendingLimitGas from '@/hooks/useSpendingLimitGas'
+import AdvancedParams, { useAdvancedParams } from '@/components/tx/AdvancedParams'
+import useChainId from '@/hooks/useChainId'
+import { useWeb3 } from '@/hooks/wallets/web3'
+import { parseUnits } from '@ethersproject/units'
+import { EMPTY_DATA, ZERO_ADDRESS } from '@gnosis.pm/safe-core-sdk/dist/src/utils/constants'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { Errors, logError } from '@/services/exceptions'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import { useCurrentChain } from '@/hooks/useChains'
+import { dispatchSpendingLimitTxExecution } from '@/services/tx/txSender'
+import { getTxOptions } from '@/utils/transactions'
+
+export type SpendingLimitTxParams = {
+  safeAddress: string
+  token: string
+  to: string
+  amount: BigNumberish
+  paymentToken: string
+  payment: BigNumberish
+  delegate: string
+  signature: BytesLike
+}
+
+const ReviewSpendingLimitTx = ({ params, onSubmit }: ReviewTokenTxProps): ReactElement => {
+  const [isSubmittable, setIsSubmittable] = useState<boolean>(true)
+  const [submitError, setSubmitError] = useState<Error | undefined>()
+  const chainId = useChainId()
+  const currentChain = useCurrentChain()
+  const provider = useWeb3()
+  const { safeAddress } = useSafeInfo()
+  const { balances } = useBalances()
+  const token = balances.items.find((item) => item.tokenInfo.address === params.tokenAddress)
+  const spendingLimit = useSpendingLimit(token?.tokenInfo)
+
+  const txParams: SpendingLimitTxParams = useMemo(
+    () => ({
+      safeAddress,
+      token: spendingLimit?.token || ZERO_ADDRESS,
+      to: params.recipient,
+      amount: parseUnits(params.amount, token?.tokenInfo.decimals).toString(),
+      paymentToken: ZERO_ADDRESS,
+      payment: 0,
+      delegate: spendingLimit?.beneficiary || ZERO_ADDRESS,
+      signature: EMPTY_DATA,
+    }),
+    [
+      params.amount,
+      params.recipient,
+      safeAddress,
+      spendingLimit?.beneficiary,
+      spendingLimit?.token,
+      token?.tokenInfo.decimals,
+    ],
+  )
+
+  const { gasLimit, loading } = useSpendingLimitGas(txParams)
+
+  const [advancedParams, setManualParams] = useAdvancedParams({ gasLimit })
+
+  const handleSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault()
+    setIsSubmittable(false)
+    setSubmitError(undefined)
+
+    if (!provider) return
```

Comment:
This should probably be right after preventDefault otherwise the button will be disabled forever.